### PR TITLE
Adds rotate path operation

### PR DESCRIPTION
**Pull request by gquito**
#45 #36 

### Here's an example of rotation of about the center of this sword by 90° .
image
image

### CODE REVIEW

**Comment by kuubeu** [comment]
Nice! 
A couple of notes from me:
- I think using something like "Origin X/Y" instead of "Rotate X/Y" and "Angle" instead of "Degree" in the input fields would be easier to understand (maybe the degree symbol (°) would append to the input?)
- If we changed the grid from `6fr 6fr 7fr` to `1fr 1fr 1fr` you could use `2fr 2fr 2fr 3fr` instead, leaving a bit more space for the input fields:
path-operations-alternative
- It would be nice if horizontal and vertical lines were kept or converted to the appropriate ones when rotating by multiples of 90°, instead of always being converted to lines

File: src/app/app.component.ts [hunk 1]
```diff
@@ -322,6 +325,14 @@ export class AppComponent implements AfterViewInit {
     this.afterModelChange();
   }
 
+  rotate(x: number, y: number, angle: number) {
+    this.parsedPath.rotate(1 * x, 1 * y, 1 * angle);
+    this.rotateX = x;
+    this.rotateY = y;
+    this.rotateAngle = angle;
```
Comment (Yqnn) [inline src/app/app.component.ts:332]: The 3 above lines do nothing, and could be removed.

**Comment by Yqnn** [comment]
Thanks a lot @gquito  for this very nice change 👍 
I agree with the 3 remarks from @kuubeu, could you please fix at least the 2 first ones ? Then I will merge the PR.

**Comment by kuubeu** [comment]
Awesome! 🤍

**Pull request merged**